sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -106,21 +106,27 @@ private[hive] class HiveClientImpl(
 
     // Set up kerberos credentials for UserGroupInformation.loginUser within
     // current class loader
-    // Instead of using the spark conf of the current spark context, a new
-    // instance of SparkConf is needed for the original value of spark.yarn.keytab
-    // and spark.yarn.principal set in SparkSubmit, as yarn.Client resets the
-    // keytab configuration for the link name in distributed cache
-    if (sparkConf.contains("spark.yarn.principal") && sparkConf.contains("spark.yarn.keytab")) {
+    if (sparkConf.contains("spark.yarn.principal")) {
```

Comment:
I see. Man this is messy. Is there a better way to check where this code is running so that you don't have to to fs checks to see if the files exist? There's a very unlikely possibility that on the AM side "spark.yarn.keytab" might point at a real file that is not the desired keytab...

I also noticed that in the client case, it doesn't seem like delegation token renewal will work. I'm not sure how well that works in general in client mode anyway, but might be something worth tracking in a separate bug.